ui/src/app/pages/login/sign-in.tsx

```diff
@@ -414,6 +416,9 @@ export class SignInImpl extends React.Component<SignInProps, SignInState> {
   }
 
   private onSubmit = async () => {
+    this.setState({
+      loading: true,
+    });
```

Comment:
You could continue to reinforce the pattern of only using the "update" version of set state by leveraging lodash here:
```suggestion
    this.setState(fp.set('loading', true));
```

Reply:
Could you please elaborate on this? I thought that
```
this.setState({
      loading: true,
 });
```
would take the existing state and only update the loading attribute. Does the fp.set(...) do something different?